propolis/src/instance.rs

```diff
@@ -408,6 +408,11 @@ impl Instance {
 
             next_state = state.next_transition(inner.state_target);
         }
+
+        // Explicitly drop the instance-held reference to the Machine.  If all
+        // the worker tasks (sync or async) are complete, this should be the
+        // last holder and result in its destruction.
+        let _ = inner.machine.take();
```

Comment:
Why do we need to do this explicitly? Forcing all access to the machine to go through an extra `unwrap` is kinda gross - wouldn't this happen implicitly as the last reference to the instance goes out of scope?

Reply:
This is a bit complicated (and perhaps would benefit from a longer comment) : I need the `Machine` to get dropped before the `Instance`, so the `mmap()` areas it owns are `munmap()`ed prior to the `Instance` doing a destroy of the vmm handle.